geopyspark/geotrellis/catalog.py

```diff
@@ -253,53 +83,68 @@ def read_value(layer_type,
         zdt (``datetime.datetime``): The time stamp of the tile if the data is spatial-temporal.
             This is represented as a ``datetime.datetime.`` instance.  The default value is,
             ``None``. If ``None``, then only the spatial area will be queried.
-        options (dict, optional): Additional parameters for reading the tile for specific backends.
-            The dictionary is only used for ``Cassandra`` and ``HBase``, no other backend requires
-            this to be set.
-        **kwargs: The optional parameters can also be set as keywords arguments. The keywords must
-            be in camel case. If both options and keywords are set, then the options will be used.
 
     Returns:
         :class:`~geopyspark.geotrellis.Tile`
     """
 
-    if not _in_bounds(layer_type, uri, layer_name, layer_zoom, col, row):
-        return None
+    if uri in _cached_stores:
+        store = _cached_stores[uri]
     else:
-        options = options or kwargs or {}
+        store = AttributeStore.build(uri)
+        _cached_stores[uri] = store
 
-        if zdt:
-            zdt = zdt.astimezone(pytz.utc).isoformat()
-        else:
-            zdt = ''
+    reader = ValueReader(uri, layer_name, layer_zoom, store)
```

Comment:
I am not sure, but I don't think so. The main cost we need to cache is the layer metadata lookup. We need to avoid fetching it on every tile request. Having an instance of `AttributeStoreWrapper` should take care of that. 

All the remaining tile readers are basically just interface class creations leading to `.read` call. I could see the python caching machinery taking longer to work then the actual JVM class creation. So I decided to put this question off until we're benchmarking the tile server.